src/decorators/sub-store.spec.ts

```diff
@@ -0,0 +1,243 @@
+import { NgZone, Component, Injectable } from '@angular/core';
```

Comment:
Even though I'm not the biggest fan of inheritance - should we have some test cases around components that extend other components/classes?